### PR TITLE
[MNT] Vector Classifiers from _contrib to Classification Package

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Removing stuff from _contrib and moving classifiers to the classification package (#2942, #2947).

~~Removed files:~~
- ~~classification_checklist~~
- ~~temporal_importance_diagram~~

Moved files:
- _classification_test_reproduction to classification/tests
- _rotation_forest to classification_sklearn
- _continuous_interval_tree to classification_sklearn

### CODE REVIEW

**Comment by TonyBagnall** [comment]
@fkiraly could you clarify the changes you are requesting here please?

**Review by fkiraly — CHANGES_REQUESTED** [review]
As in #2947, citing from there:

These are `sklearn` compatible classifiers, correct?
So, I would put them into a `classification_sklearn` dir, and ensure we have tests in there which run the `sklearn` `check_estimator` on the classifiers.

File: sktime/classification_sklearn/__init__.py
Comment (aiwalter) [inline sktime/classification_sklearn/__init__.py:1]: not sure if `classification_sklearn` is the best name, but its not a blocker for me

**Review by aiwalter — DISMISSED** [review]

File: sktime/classification_sklearn/__init__.py
Comment (fkiraly) [reply sktime/classification_sklearn/__init__.py:1]: well, any better ideas for the name of the folder or location?
I just think it should be in some dedicate location where sklearn classifiers are

**Comment by fkiraly** [comment]
Question, where has `temporal_importance_diagram` gone?

**Comment by MatthewMiddlehurst** [comment]
To the ether? I'm not sure where deleted files go.

As to why I elected to remove it, it's a visualisation tool very specific to interval based classifiers. I'm pretty sure we have no area for visualisation, and I'm not willing to tidy up the file to put it into sktime proper. If someone else wants to take on the task, be my guest. It is however in _contrib and has to go somewhere.

**Comment by TonyBagnall** [comment]
@fkiraly could you clarify what changes you are requesting, or would you in fact like to keep contrib but with redundant material removed? @MatthewMiddlehurst can you put back temporal_importance_diagram for now so we can at least get the critical files moved and this PR approved? moving classifiers and removing content are separate issues, the former hopefully less likely to be blocked

**Comment by MatthewMiddlehurst** [comment]
Added back in removed files. sklearn classifiers are in their own directory and have tests now.

**Comment by TonyBagnall** [comment]
@fkiraly could you clarify what changes, if any, you want to this PR?

**Review by TonyBagnall — DISMISSED** [review]

**Comment by TonyBagnall** [comment]
@fkiraly could you either unblock this PR or clarify what changes, if any, you want?

**Comment by fkiraly** [comment]
>@fkiraly could you either unblock this PR or clarify what changes, if any, you want?

@TonyBagnall, I'm not sure what you are asking for, imo "what changes" has been clarified above, and you have asked at least three times after that.
I have written a general list of suggestions in https://github.com/alan-turing-institute/sktime/issues/2942, and I have replied in your duplicate issue https://github.com/alan-turing-institute/sktime/issues/2947 that you closed, also with concrete asks.

It is ok if you disagree, but then please comment on what I wrote rather than just repeating your request for me to "please clarify". This seems to imply that I have said nothing, but, as you can see if you look above and/or follow the links I just provided, I have.

If asking for clarification, kindly be constructive, comment on what I already said, and specify what you need clarification on.

The general protocol for PR is also for the person who has opened the PR (here: @MatthewMiddlehurst) to react to change requests, in the form of "addressed", or "please clarify [specific item]", or "I do not think relevant or good idea because [reason]". When considering the change requests addressed, a re-review should be requested by the reviewer, through the GitHub interface.

Here are some useful resources on how PR reviews and the interaction around them works:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests
https://google.github.io/eng-practices/review/developer/handling-comments.html

**Comment by fkiraly** [comment]
Maybe to clarify: we do not need to make this PR scope identical with #2942, we can split in multiple PR or discuss what we do. Still, my suggestions apply as long as intended scope by @MatthewMiddlehurst and the scope of #2942 intersect.

File: sktime/classification_sklearn/__init__.py
Comment (fkiraly) [reply sktime/classification_sklearn/__init__.py:1]: I'm fine with either location as long as there is a clear naming convention and rationale.

File: sktime/classification/tests/test_all_classifiers.py
Comment (fkiraly) [inline sktime/classification/tests/test_all_classifiers.py:107]: this was a bug, correct?

For robustness, kindly use
```python
        _, _, X_train_metadata = check_is_scitype(X_train, "Panel", return_metadata=True)
        X_train_len = X_train_metadata["n_instances"]
```

This should ensure the test is still correct in case the mtype of `X_train` changes

Comment (MatthewMiddlehurst) [reply sktime/classification/tests/test_all_classifiers.py:112]: Done

File: sktime/classification_sklearn/__init__.py
Comment (MatthewMiddlehurst) [reply sktime/classification_sklearn/__init__.py]: Package has changed to classification_sklearn to classification.sklearn

**Review by TonyBagnall — APPROVED** [review]

**Comment by TonyBagnall** [comment]
this PR is to move two classifiers out of contrib. The scope of #2942 is much wider, and I dont think it is necessary myself. I would vote for just deleting contrib. But if you or anyone else wants to salvage some of the code there feel free. I do not feel strongly about it, but I do think these classifiers should move, so can we just get this PR in please?

**Comment by fkiraly** [comment]
> If asking for clarification, kindly be constructive, comment on what I already said, and specify what you need clarification on.

> The general protocol for PR is also for the person who has opened the PR (here: @MatthewMiddlehurst) to react to change requests, in the form of "addressed", or "please clarify [specific item]", or "I do not think relevant or good idea because [reason]". When considering the change requests addressed, a re-review should be requested by the reviewer, through the GitHub interface.

**Comment by MatthewMiddlehurst** [comment]
I have addressed the above change requests, this should be good to go?

**Review by TonyBagnall — APPROVED** [review]

**Comment by fkiraly** [comment]
> but I do think these classifiers should move, so can we just get this PR in please?

Please remain calm, @TonyBagnall, and please respect the work that core developers are doing here.
There is no entitlement to get a review, and a couple days are a normal duration for getting a reply on a PR.

**Review by fkiraly — APPROVED** [review]
Looks good.

**Pull request merged**
